### PR TITLE
Remove unused Memcpy methods from shared Buffer implementation

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: src/System.Private.CoreLib/src/System/StubHelpers.cs [hunk 1]
```diff
@@ -78,18 +78,29 @@ internal static unsafe IntPtr ConvertToNative(int flags, string strManaged, IntP
...
             {
-                // Otherwise we use a slower "2-pass" mode where we first marshal the string into an intermediate buffer
-                // (managed byte array) and then allocate exactly the right amount of unmanaged memory. This is to avoid
-                // wasting memory on systems with multibyte character sets where the buffer we end up with is often much
-                // smaller than the upper bound for the given managed string.
+                if (strManaged.Length == 0)
+                {
+                    nb = 0;
+                    pbNativeBuffer = (byte*)Marshal.AllocCoTaskMem(2);
+                }
```
Comment (stephentoub) [inline src/System.Private.CoreLib/src/System/StubHelpers.cs]: Where did this special case come from?

Comment (stephentoub) [reply src/System.Private.CoreLib/src/System/StubHelpers.cs]: Oh, I see, it's a translation of the `if (len == 0) return;` in Memcpy.

File: src/System.Private.CoreLib/shared/System/Buffer.cs [hunk 2]
```diff
@@ -112,34 +112,6 @@ public static unsafe void MemoryCopy(void* source, void* destination, ulong dest
             Memmove((byte*)destination, (byte*)source, checked((nuint)sourceBytesToCopy));
         }
 
-        internal static unsafe void Memcpy(byte[] dest, int destIndex, byte* src, int srcIndex, int len)
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Buffer.cs]: VM depends on these methods as well here: https://github.com/dotnet/coreclr/blob/master/src/vm/mscorlib.h#L937

Comment (marek-safar) [reply src/System.Private.CoreLib/shared/System/Buffer.cs]: yeah (moved it to .CoreCLR.cs version)

**Comment by jkotas** [comment]
@jkoritzinsky Could you please review the IL marshaller changes?

**Pull request merged**